qiskit/dagcircuit/dagcircuit.py

```diff
@@ -2090,7 +2090,7 @@ def layers(self, *, vars_mode: _VarsMode = "captures"):
             new_layer = self.copy_empty_like(vars_mode=vars_mode)
 
             for node in op_nodes:
-                new_layer._apply_op_node_back(node, check=False)
+                new_layer.apply_operation_back(node.op, node.qargs, node.cargs, check=False)
```

Comment:
This is a potential transpiler performance regression in the 1.2 branch, but I think it'll only affect `StochasticSwap`, so not a big deal.  If you feel like it, it _might_ be worth doing a shallow copy on `node` before calling `_apply_op_node_back`, and see if that's sufficient to fix it as well.

Reply:
(since I'd guess the problem is the inner mutation of the `_node_id` field - it won't be shared between the source and destination DAGs)